### ISSUE TITLE
Light cache file write error: Permission denied (os error 13)

### ISSUE THREAD

**Issue by jameshilliard** [paritytech/parity#5805]
I'm seeing this error when running parity with --base-path under a user with heavily restricted permissions. It would appear that parity is trying to write the cache file to some directory other than what is set using --base-path and failing since it is not allowed access to whatever path it is attempting to use.

Confirmed with strace that it is trying to use a non-existing home folder even though --base-path is set to /var/lib/parity:
```
[pid 101910] stat("/home/parity/.ethash/light/c67d12cb817cfe2de207e4823f6fa22ac14b924c3072e1ccc9b7718329287d1c", 0x7f760abf9e10) = -1 ENOENT (No such file or directory)
[pid 101910] stat("/home/parity/.ethash/light", 0x7f760abf9d10) = -1 ENOENT (No such file or directory)
[pid 101910] stat("/home/parity/.ethash", 0x7f760abf9c40) = -1 ENOENT (No such file or directory)
[pid 101910] stat("/home/parity", 0x7f760abf9b70) = -1 ENOENT (No such file or directory)
[pid 101910] stat("/home", {st_mode=S_IFDIR|0755, st_size=16, ...}) = 0
[pid 101910] mkdir("/home/parity", 0777) = -1 EACCES (Permission denied)
2017-06-09 13:22:47  Light cache file write error: Permission denied (os error 13)
```

**Comment by 5chdn** [comment]
@tomusdrw will address this in #4565 - please subscribe to follow progress.

**Comment by etscrivner** [comment]
This is probably worth getting a patch release out for. Still happening on 1.7.0 and causing some issues her at Coinbase.

**Comment by 5chdn** [comment]
Hi @etscrivner, #5881 is included in 1.7.0 - could you share the full logs here, so we can confirm it's the exact same issue?

**Comment by roninkaizen** [comment]
I'm running:

        Parity version: 1.7.0
        Operating system: Linux mint 18 Sarah-
 uname -r = 4.4.0-93-generic
        And installed: via deb on 2017.09.09
  version Parity/v1.7.0-beta-5f2cabd-20170727/x86_64-linux-gnu/rustc1.18.0

although not being asked- i respond myself onto this as follows
2017-09-09 17:22:28  Starting Parity/v1.7.0-beta-5f2cabd-20170727/x86_64-linux-gnu/rustc1.18.0
2017-09-09 17:22:28  Keys path /home/mint/.local/share/io.parity.ethereum/keys/Foundation
2017-09-09 17:22:28  DB path /home/mint/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2017-09-09 17:22:28  Path to dapps /home/mint/.local/share/io.parity.ethereum/dapps
2017-09-09 17:22:28  State DB configuration: fast
2017-09-09 17:22:28  Operating mode: active
2017-09-09 17:22:28  Configured for Foundation using Ethash engine
<----->
2017-09-09 17:28:52  Light cache file write error: Permission denied (os error 13)
2017-09-09 17:29:24  Light cache file write error: Permission denied (os error 13)
2017-09-09 17:29:29  Light cache file write error: Permission denied (os error 13)
<----->
2017-09-09 17:31:03  Light cache file write error: Permission denied (os error 13)
2017-09-09 17:31:26  Light cache file write error: Permission denied (os error 13)
2017-09-09 17:31:34  Light cache file write error: Permission denied (os error 13)
2017-09-09 17:31:53  Light cache file write error: Permission denied (os error 13)
2017-09-09 17:32:08  Light cache file write error: Permission denied (os error 13)
<----->
2017-09-10 19:03:53  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:08:16  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:18:50  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:20:21  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:20:48  Light cache file write error: Permission denied (os error 13)
<----->
2017-09-10 19:24:29  Error writing light request timing samples to file: 
IoError: Permission denied (os error 13)
<----->
2017-09-10 19:28:04  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:29:54  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:30:32  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:30:42  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:31:42  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:34:31  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:35:40  Light cache file write error: Permission denied (os error 13)
2017-09-10 19:36:05  Light cache file write error: Permission denied (os error 13)

Has no influence on stability.

**Comment by 5chdn** [comment]
@tomusdrw can you comment on this?

**Comment by tomusdrw** [comment]
`$BASE_PATH/cache` should be writeable. @roninkaizen Could you run with `strace` and check what path it expects to write the cache files to?

**Comment by roninkaizen** [comment]
@tomusdrw ->still learning
avail-@-> https://github.com/roninkaizen/recipes/blob/master/trace.log.170

**Comment by tomusdrw** [comment]
@roninkaizen The strace you published contained some sensitive data - I removed the comment. Could you please run it for some more and publish only relevant part maybe `| grep cache`

**Comment by etscrivner** [comment]
@tomusdrw Here is the relevant information. It looks like a slightly different error from the one above:

Configuration Flags:

```
--no-download --base-path /opt/parity/data --chain homestead
```

Strace Results:

```
mkdir("/opt/parity/data", 0777)         = -1 EEXIST (File exists)
stat("/opt/parity/data", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
mkdir("/opt/parity/data/chains", 0777)  = -1 EEXIST (File exists)
stat("/opt/parity/data/chains", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
mkdir("$BASE/cache", 0777)              = -1 ENOENT (No such file or directory)
mkdir("$BASE", 0777)                    = -1 EACCES (Permission denied)
stat("$BASE", 0x7ffdb3935b90)           = -1 ENOENT (No such file or directory)
write(2, "Permission denied (os error 13)", 31Permission denied (os error 13)) = 31
```

**Comment by tomusdrw** [comment]
The issue was addressed in #6212 and will be part of 1.7.2.
